src/config_parse/config.cpp

```diff
@@ -0,0 +1,89 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   Config.cpp                                         :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: lizhang <lizhang@student.codam.nl>           +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2023/06/26 16:49:22 by lizhang       #+#    #+#                 */
+/*   Updated: 2023/06/29 16:17:02 by lizhang       ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../include/Config.hpp"
+#include <iostream>
+#include <stdlib.h>
+
+
+std::string Config::readConfigFile(std::string name)
+{
+	std::ifstream	file;
+	std::string		file_content;
+
+	file.open(file_name);
+	if(file.is_open() == 0)
+		throw(std::invalid_argument(strerror(errno)));
+	getline(file, file_content, '\0');
+	if (file_content.length() <= 1)
+		throw(std::invalid_argument("No content in config file"));
+	file.close();
+	return(file_content);
+}
+
+Server	newServer(std::string settings)
+{
+	std::string 				server_name;
+	DirSettings					root;
+	std::vector<int>			ports;
+	std::vector<DirSettings>	optional;
```

Comment:
Could you add one more element to store the information for rootDir?
`std::string rootDir`